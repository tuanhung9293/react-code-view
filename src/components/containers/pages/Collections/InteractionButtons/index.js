import React from 'react'
import {connect} from 'react-redux'
import LikeButton from 'components/ui/buttons/LikeButton'
import DeleteButton from 'components/ui/buttons/DeleteButton'
import EditButton from 'components/ui/buttons/EditButton'
import {likeCollection, removeCollection} from 'state/actions/collectionsAction'
import styles from './InteractionButtons.module.css'

const InteractionButtons = ({itemData, likedCollection, deleteCollection}) => {
  return (
    <div className={styles.base}>
      <LikeButton color={`${itemData.liked ? 'red' : 'white'}`} onClick={() => likedCollection(itemData.nasaId)} />
      <DeleteButton onClick={() => deleteCollection(itemData.nasaId)} />
      <EditButton to={`/collection/edit/${itemData.nasaId}`} />
    </div>
  )
}

export default connect(
  null,
  (dispatch) => ({
    likedCollection: (collectionId) => dispatch(likeCollection(collectionId)),
    deleteCollection: (collectionId) => dispatch(removeCollection(collectionId))
  })
)(InteractionButtons)
