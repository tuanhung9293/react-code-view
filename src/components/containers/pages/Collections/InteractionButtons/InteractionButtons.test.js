import React from 'react'
import {mount, shallow} from 'enzyme'
import {MemoryRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import InteractionButtons from './'

import configureStore from 'redux-mock-store'
const mockStore = configureStore()

let store = mockStore({
  medias: {
    collections: {
      data: [{}, {}, {}]
    }
  }
})
store.dispatch = () => {}


describe('renders', () => {
  it('renders InteractionButtons connected', () => {
    expect(shallow(
      <Provider store={store}>
        <InteractionButtons />
      </Provider>
    )).toMatchSnapshot()
  })

  it('renders InteractionButtons Component', () => {
    jest.mock('../../../../ui/buttons/Button', () => ()=> <div id="mockButton">
      Button
    </div>)

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[{pathname: '/', key: 'testKey'} ]}>
          <InteractionButtons
            itemData={{liked: true, nasaId: 123}}
            likedCollection={() => {}}
            deleteCollection={() => {}}
          />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(InteractionButtons)).toMatchSnapshot()
  })
})

