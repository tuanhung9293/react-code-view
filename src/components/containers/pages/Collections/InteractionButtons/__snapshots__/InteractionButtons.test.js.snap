// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders InteractionButtons Component 1`] = `
<Connect(InteractionButtons)
  deleteCollection={[Function]}
  itemData={
    Object {
      "liked": true,
      "nasaId": 123,
    }
  }
  likedCollection={[Function]}
>
  <InteractionButtons
    deleteCollection={[Function]}
    itemData={
      Object {
        "liked": true,
        "nasaId": 123,
      }
    }
    likedCollection={[Function]}
  >
    <div
      className="base"
    >
      <LikeButton
        color="red"
        onClick={[Function]}
      >
        <IconButton
          onClick={[Function]}
        >
          <Button
            className="button"
            onClick={[Function]}
          >
            <button
              className="button"
              onClick={[Function]}
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 4.419c-2.826-5.695-11.999-4.064-11.999 3.27 0 7.27 9.903 10.938 11.999 15.311 2.096-4.373 12-8.041 12-15.311 0-7.327-9.17-8.972-12-3.27z"
                  fill="red"
                  stroke="black"
                />
              </svg>
            </button>
          </Button>
        </IconButton>
      </LikeButton>
      <DeleteButton
        onClick={[Function]}
      >
        <IconButton
          onClick={[Function]}
        >
          <Button
            className="button"
            onClick={[Function]}
          >
            <button
              className="button"
              onClick={[Function]}
            >
              <img
                alt="Add to favorite"
                src="test-file-stub"
              />
            </button>
          </Button>
        </IconButton>
      </DeleteButton>
      <EditButton
        to="/collection/edit/123"
      >
        <IconButton
          to="/collection/edit/123"
        >
          <Button
            className="button"
            to="/collection/edit/123"
          >
            <Link
              className="button"
              replace={false}
              to="/collection/edit/123"
            >
              <a
                className="button"
                href="/collection/edit/123"
                onClick={[Function]}
              >
                <img
                  alt="Edit"
                  src="test-file-stub"
                />
              </a>
            </Link>
          </Button>
        </IconButton>
      </EditButton>
    </div>
  </InteractionButtons>
</Connect(InteractionButtons)>
`;

exports[`renders renders InteractionButtons connected 1`] = `<Connect(InteractionButtons) />`;
